doc/source/text.rst

```diff
@@ -201,9 +207,92 @@ and optional groups like
 
 .. ipython:: python
 
-   pd.Series(['a1', 'b2', '3']).str.extract('(?P<letter>[ab])?(?P<digit>\d)')
+   pd.Series(['a1', 'b2', '3']).str.extract('([ab])?(\d)')
+
+can also be used. Note that any capture group names in the regular
+expression will be used for column names; otherwise capture group
+numbers will be used.
+
+Extracting a regular expression with one group returns a DataFrame
+with one column if ``expand=True``, and a Series if ``expand=False``.
+
+.. ipython:: python
+
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=True)
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=False)
+
+Calling ``index.str.extract`` on a regex with exactly one capture
```

Comment:
I would say calling on an ``Index`` (e.g.don't use ``index.str.extract``) with ``.str.extract``. Then this language is the same as above.